tests/ui/lazy_lock_like/auxiliary/once_cell.rs

```diff
@@ -0,0 +1,52 @@
+//! **FAKE** once_cell crate.
+
+pub mod sync {
+    use core::cell::Cell;
+    use std::marker::PhantomData;
+
+    pub struct OnceCell<T>(PhantomData<T>);
+    impl<T> OnceCell<T> {
+        pub const fn new() -> OnceCell<T> {
+            OnceCell(PhantomData)
+        }
+    }
+    impl<T> Default for OnceCell<T> {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+
+    pub struct Lazy<T, F = fn() -> T> {
+        cell: OnceCell<T>,
+        init: Cell<Option<F>>,
+    }
```

Comment:
Remove `OnceCell` entirely and replace it with `PhantomData` here.

Even just this might work: 

```suggestion
    pub struct Lazy<T, F = fn() -> T> {
        cell: T,
        init: F,
    }
```